src/server/replica.cc

```diff
@@ -733,75 +771,42 @@ error_code Replica::ParseAndExecute(base::IoBuf* io_buf) {
-    io::Result<size_t> eof_res = chained.Read(io::MutableBytes{buf.get(), eof_token.size()});
-    if (!eof_res || *eof_res != eof_token.size()) {
-      LOG(ERROR) << "Error finding eof token in the stream";
-    }
+Replica::Info Replica::GetInfo() const {
+  CHECK(sock_);
 
-    // TODO - to compare tokens
-  }
-  VLOG(1) << "ReplicateDFFb finished after reading " << loader.bytes_read() << " bytes";
+  return sock_->proactor()->AwaitBrief([this] {
+    Info res;
+    res.host = master_context_.host;
+    res.port = master_context_.port;
+    res.master_link_established = (state_mask_ & R_TCP_CONNECTED);
+    res.sync_in_progress = (state_mask_ & R_SYNCING);
+    res.master_last_io_sec = (ProactorBase::GetMonotonicTimeNs() - last_io_time_) / 1000000000UL;
+    return res;
+  });
 }
 
-error_code Replica::StartFlow() {
-  CHECK(!sock_);
-  DCHECK(!master_context_.master_repl_id.empty() && !master_context_.dfly_session_id.empty());
-
-  ProactorBase* mythread = ProactorBase::me();
-  CHECK(mythread);
-
-  sock_.reset(mythread->CreateSocket());
-  RETURN_ON_ERR(sock_->Connect(master_context_.master_ep));
-
-  VLOG(1) << "Sending on flow " << master_context_.master_repl_id << " "
-          << master_context_.dfly_session_id << " " << master_context_.flow_id;
-
-  ReqSerializer serializer{sock_.get()};
-  serializer.SendCommand(StrCat("DFLY FLOW ", master_context_.master_repl_id, " ",
-                                master_context_.dfly_session_id, " ", master_context_.flow_id));
-  RETURN_ON_ERR(serializer.ec());
-
-  parser_.reset(new RedisParser{false});  // client mode
-
-  std::unique_ptr<base::IoBuf> io_buf{new base::IoBuf(128)};
-  unsigned consumed = 0;
-  RETURN_ON_ERR(ReadRespReply(io_buf.get(), &consumed));  // uses parser_
+bool Replica::CheckRespIsSimpleReply(string_view reply) const {
+  return resp_args_.size() == 1 || resp_args_.front().type == RespExpr::STRING ||
+         ToSV(resp_args_.front().GetBuf()) == reply;
+}
 
-  if (resp_args_.size() < 2 || resp_args_[0].type != RespExpr::STRING ||
-      resp_args_[1].type != RespExpr::STRING) {
-    LOG(ERROR) << "Bad FLOW response " << ToSV(io_buf->InputBuffer());
-    return make_error_code(errc::bad_message);
+bool Replica::CheckRespFirstTypes(initializer_list<RespExpr::Type> types) const {
+  unsigned i = 0;
+  for (RespExpr::Type type : types) {
+    if (i >= resp_args_.size() || resp_args_[i].type != type)
+      return false;
+    ++i;
   }
+  return true;
+}
 
-  string_view flow_directive = ToSV(resp_args_[0].GetBuf());
-  string eof_token;
-  if (flow_directive == "FULL") {
-    eof_token = ToSV(resp_args_[1].GetBuf());
-  } else {
-    LOG(ERROR) << "Bad FLOW response " << ToSV(io_buf->InputBuffer());
+error_code Replica::SendCommand(string_view command, ReqSerializer* serializer) {
+  serializer->SendCommand(command);
+  error_code ec = serializer->ec();
+  if (!ec) {
+    last_io_time_ = sock_->proactor()->GetMonotonicTimeNs();
   }
```

Comment:
`last_io_time_` is last successful?

Reply:
yeah, as fart as I remember, it's for `INFO` - the replication section. When we last heard from the master.